src/main/java/io/jenkins/plugins/services/impl/GithubContentsExtractor.java

```diff
@@ -38,15 +48,47 @@ public String getRepo() {
     }
   }
 
-  private static final Pattern REPO_PATTERN = Pattern
-      .compile("https?://github.com/jenkinsci/([^/.]+)/blob/([^/]+)/(.+\\.(md|adoc))$");
-  
-  private static final String CONTENTS_ENDPOINT = "contents/%s";
+  private static final Pattern REPO_PATTERN = Pattern.compile("https?://github.com/jenkinsci/([^/.]+)/blob/([^/]+)/(.+\\.(md|adoc))$");
```

Comment:
I may be missing it or maybe it's not implemented yet but how does it work if the url is like:
`https://github.com/jenkinsci/blueocean-plugin` and doesn't include the full file name? or does that get looked up somewhere else?

Reply:
https://github.com/jenkins-infra/plugin-site-api/blob/master/src/main/java/io/jenkins/plugins/services/impl/GithubReadmeExtractor.java

the extractors take a bit of time to figure out which get applied to which

Reply:
Do you plan to cover the short URL cases too? I think they are quite common.

Reply:
Is there a test case or example of those?
if by short url cases, you mean just the root, thats the readme extractor

Reply:
> thats the readme extractor

Exactly, but that one's probably used more than the content extractor , so if you want to get rid of API usage it would make sense to change that also (maybe fetch the README urls by GraphQL, 1 query for 100 plugins, then use Markdown/ AsciiDoc magic for raw content).

But I'm not sure if this PR is still relevant or if the plan is to move all the functionality to `plugin-site` and/or `update-center2`.

Reply:
> Exactly, but that one's probably used more than the content extractor

Yea, I think more people should use the full path in <url> but this still helps.

I really don't know though, thats why i marked it as WIP cause i wanted feedback.